dev-java/dom4j/dom4j-2.1.3.ebuild

```diff
@@ -0,0 +1,66 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+# Skeleton command:
+# java-ebuilder --generate-ebuild --workdir . --pom dom4j-2.1.3.pom --download-uri https://github.com/dom4j/dom4j/archive/refs/tags/version-2.1.3.tar.gz --slot 1 --keywords "~amd64 ~arm ~arm64 ~ppc64 ~x86" --ebuild dom4j-2.1.3.ebuild
+
+EAPI=7
+
+JAVA_PKG_IUSE="doc source test"
+MAVEN_ID="org.dom4j:dom4j:2.1.3"
+JAVA_TESTING_FRAMEWORKS="junit-4"
+
+inherit java-pkg-2 java-pkg-simple
+
+DESCRIPTION="flexible XML framework for Java"
+HOMEPAGE="https://dom4j.github.io/"
+SRC_URI="https://github.com/${PN}/${PN}/archive/refs/tags/version-${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="dom4j"
+SLOT="1"
+KEYWORDS="~amd64 ~arm ~arm64 ~ppc64 ~x86"
+IUSE="jaxen"
+
+CDEPEND="
+	dev-java/jaxb-api:0
+	dev-java/xpp2:0
+	dev-java/xpp3:0
+	dev-java/xsdlib:0
+	jaxen? ( dev-java/jaxen:1.2[dom4j] )
+"
+DEPEND="${CDEPEND}
+	>=virtual/jdk-1.8:*"
+
+# Runtime dependencies
+# POM: ${P}.pom
+# javax.xml.bind:jaxb-api:2.2.12 -> !!!groupId-not-found!!!
+# javax.xml.stream:stax-api:1.0-2 -> java-virtuals/stax-api:0
+# jaxen:jaxen:1.1.6 -> >=dev-java/jaxen-1.2.0:1.2
+# net.java.dev.msv:xsdlib:2013.6.1 -> >=dev-java/xsdlib-20090415:0
+# pull-parser:pull-parser:2 -> !!!groupId-not-found!!!
+# xpp3:xpp3:1.1.4c -> >=dev-java/xpp3-1.1.4c:0
+
+RDEPEND="${CDEPEND}
+	>=virtual/jre-1.8:*
+"
+
+S="${WORKDIR}/${PN}-version-${PV}"
+
+JAVA_GENTOO_CLASSPATH="jaxb-api,xpp2,xpp3,xsdlib"
+JAVA_SRC_DIR="src/main/java"
+
+JAVA_TEST_GENTOO_CLASSPATH="junit-4"
+JAVA_TEST_SRC_DIR="src/test/java"
+
+src_prepare() {
+	default
+
+	# According to README.md the dependency on Jaxen is optional
+	# Break cirular deps by switching it off.
+
+	if use jaxen; then
+		JAVA_GENTOO_CLASSPATH+=" jaxen-1.2"
+#	else
+#	N E E D  H E L P
```

Comment:
i recall when reading about java modules, there was the optional feature mentioned too. what i got from that is that it's optional only during runtime, not during compilation. [here](https://maven.apache.org/guides/introduction/introduction-to-optional-and-excludes-dependencies.html) it's explained how optional dependencies in `pom.xml` work. in short, they change nothing for the project itself but the dependency is not transferred to another project. that's what the note in `README.md` means. so all the deps need to be provided at the compile time of `dom4j`. but there sure must be a way to patch `jaxen` out. i'll try to have a look at it when i find some time.

Reply:
> i recall when reading about java modules, there was the optional feature mentioned too. what i got from that is that it's optional only during runtime, not during compilation. [here](https://maven.apache.org/guides/introduction/introduction-to-optional-and-excludes-dependencies.html) it's explained how optional dependencies in `pom.xml` work. in short, they change nothing for the project itself but the dependency is not transferred to another project. that's what the note in `README.md` means. so all the deps need to be provided at the compile time of `dom4j`. but there sure must be a way to patch `jaxen` out. i'll try to have a look at it when i find some time.

`dom4j` seems not to use `module-info.java`.  
The way to go seems to be with [ClearDependencies](https://github.com/dom4j/dom4j/issues/99#issuecomment-830063159) as mentioned in [issue #99](https://github.com/dom4j/dom4j/issues/99).  
Got [there](https://github.com/dom4j/dom4j/issues/105) on searching the missing [XML Pull Parser](https://github.com/dom4j/dom4j/blob/master/src/main/java/org/dom4j/xpp/package-info.java) which is offered on [mvnrepository.com](https://mvnrepository.com/artifact/pull-parser/pull-parser/2.1.10) only without source code.

```
 * Compiling ...
warning: [options] bootstrap class path not set in conjunction with -source 8
src/main/java/org/dom4j/xpp/ProxyXmlStartTag.java:31: error: ProxyXmlStartTag is not abstract and does not override abstract method removeAttributeByRawName(String) in XmlStartTag
public class ProxyXmlStartTag implements XmlStartTag {
       ^
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
1 error
1 warning
```
Without sources for `XML Pull Parser` I guess we should disable that option too.

Reply:
i was not mentioning it because of modules but because of the fact that `optional` as we percieve it means something different than what seems to be the common use in java. `optional` in java usually means "still needed at compile-time but might be omitted at runtime".